src/box/schema_def.h

```diff
@@ -375,6 +376,34 @@ space_id_is_system(uint32_t space_id)
 	       space_id < BOX_SYSTEM_ID_MAX;
 }
 
+/**
+ * List of features, which are blocked until schema upgrade is done.
+ * Feature description is:
+ *   _(<token>, <number>, <major>, <minor>, <patch>)
+ * where:
+ *   token			- the enum constant for the feature, also
+ *				  used in the error message.
+ *   number			- sequential number of the feature
+ *   major, minor, patch	- version number
+ *
+ * If schema version is less than 2.11.1, then all DDL is blocked until
+ * user upgrades at least to 2.11.1. This list consists of features,
+ * which appeared after 2.11.1 and which are blocked until some version.
+ * If feature touches this file and changes schema, it should probably be
+ * added here.
+ *
+ * The only exception for now is persistent names feature. Even though
+ * they appeared in 3.0.0, we allow using them on 2.11.5 schema.
+ */
+#define BLOCKED_FEATURES(_) \
+	_(DDL_BEFORE_UPGRADE, 0, 2, 11, 1) \
+	_(PERSISTENT_NAMES, 1, 2, 11, 5) \
+	_(PERSISTENT_TRIGGERS, 2, 3, 1, 0) \
+
+ENUM(blocked_feature, BLOCKED_FEATURES);
+extern const char *blocked_feature_strs[];
+extern const struct version blocked_feature_version[];
+
```

Comment:
- Why are these defined in `schema_def.h`? I think they belong in `schema.h`, where they're used.
- I wouldn't call them "blocked" features. They aren't blocked when the schema is upgraded. Let's name them simply "schema features".
- Let's add appropriate prefixes to the constants:
  * `enum schema_feature`
  * `const char *schema_feature_strs[]`
  * `const struct version schema_feature_version[]`
  * `SCHEMA_FEATURE_DDL_BEFORE_UPGRADE`
    `SCHEMA_FEATURE_PERSISTENT_NAMES`
    `SCHEMA_FEATURE_PERSISTENT_TRIGGERS`

Reply:
Looks good, done